rlay-backend-neo4j/src/lib.rs

```diff
@@ -185,13 +319,19 @@ impl Neo4jBackend {
             deduped_cids
         };
 
-        let query = format!(
-            "MATCH (n:RlayEntity)-[r]->(m) WHERE n.cid IN {0:?} RETURN labels(n),n,type(r),m",
-            deduped_cids,
+        let query = format!("
+            UNWIND $cids AS cid
+            MATCH (n:RlayEntity {{cid: cid}})-[r]->(m)
+            RETURN labels(n),n,type(r),m"
```

Comment:
What do you think about replacing this query with the pattern comprehension query (should have same performance), which would look something like this:

```
UNWIND $cids AS cid
MATCH (n0:RlayEntity {{cid: cid}})
RETURN  {{
    self: n0,
    self_labels: labels(n0),
    children: [(n0)-[r0]->(n1) | {{
                    rel_type: type(r0),
                    self: n1,
                    self_labels: labels(n1)
    }}]
}} as data
```

That would allow us to remove `rows_to_entity` function and use the `pattern_rows` which should be easier to maintain. Plus these pattern comprehension queries can in general be generated with a future function like `patternQuery(level = 0)` where `level` corresponds to the depth of the resolve.